lib/_tls_wrap.js

```diff
@@ -275,12 +274,10 @@ function TLSSocket(socket, options) {
 
   // Wrap plain JS Stream into StreamWrap
   var wrap;
-  if (!(socket instanceof net.Socket) && socket instanceof Duplex)
-    wrap = new StreamWrap(socket);
-  else if ((socket instanceof net.Socket) && !socket._handle)
-    wrap = new StreamWrap(socket);
-  else
+  if (((socket instanceof net.Socket) && socket._handle) || !socket)
```

Comment:
linter was quiet in this case. I'm also wasn't able to find rule to check this...